sdk/core/Azure.Core/src/Cryptography/IKeyEncryptionKey.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Cryptography
+{
+    /// <summary>
+    /// A key which is used to encrypt, or wrap, another key
+    /// </summary>
+    public interface IKeyEncryptionKey
```

Comment:
I'm not an expert in this area by any stretch so I find this name a little jarring.  It's not really a key but a thing that encrypts keys, right?  Would `IKeyEncryptionClient` make sense since it follows most of our other client patterns?  Disregard if this is a well known concept/pattern that should keep using this name.

Reply:
I don't think that this should contain the word "Client" as quite often implementations of this would not call a service.